### PR TITLE
cisco-8000: qos-sai: Skip LossyQueueVoq and pgShared watermark for  t2 for time-being.

### PR DESCRIPTION
**Pull request by rraghav-cisco**
There are several testcases that are not applicable, or not work as they are written, when run on multiple-ASIC(or multi-DUT). We need to skip them for cisco-8000. These are:

1. LossyQueueVoq
           voq1 ==> split voq (not applicable for T2) or fair voq  (applicable for T2 downlink LC)
           voq2 ==> default voq so we need to add on uplink LC (need to add back)
    So we need to add it back as enhancement. 

2. PG Shared Watermark Lossy for multi-asic and multi-dut scenarion
    Reason for skip: In packet chassis lossy traffic will get dropped at egress ASIC so we cannot validate PG Watermark on ingress ASIC

### CODE REVIEW

**Comment by wangxin** [comment]
@rraghav-cisco Can you squash your commits to suppress the EasyCLA messages? BTW, your branch is out of date and may need to do a rebase.

**Comment by rraghav-cisco** [comment]
> @rraghav-cisco Can you squash your commits to suppress the EasyCLA messages? BTW, your branch is out of date and may need to do a rebase.

@wangxin : I don't have squash/merge permissions. I have rebased the code.

File: tests/qos/test_qos_sai.py [hunk 1]
```diff
@@ -540,6 +540,9 @@ def testQosSaiLosslessVoq(
             Raises:
                 RunAnsibleModuleFail if ptf test fails
         """
+        if not get_src_dst_asic_and_duts['single_asic_test']:
+             pytest.skip("This test is not applicable when "
+                 "src and dst ASICs are different.")
```
Comment (abdosi) [inline tests/qos/test_qos_sai.py:545]: update the comment here.

File: tests/qos/test_qos_sai.py [hunk 2]
```diff
@@ -1031,8 +1034,9 @@ def testQosSaiLossyQueueVoq(
             Raises:
                 RunAnsibleModuleFail if ptf test fails
         """
-        if dutTestParams["basicParams"]["sonic_asic_type"] != "cisco-8000":
-            pytest.skip("Lossy Queue Voq test is not supported")
+        if not get_src_dst_asic_and_duts['single_asic_test']:
+             pytest.skip("This test is not applicable when "
```
Comment (abdosi) [inline tests/qos/test_qos_sai.py:1038]: Remove this

File: tests/qos/test_qos_sai.py [hunk 3]
```diff
@@ -1365,6 +1369,13 @@ def testQosSaiPgSharedWatermark(
         if "wm_pg_shared_lossless" in pgProfile:
             pktsNumFillShared = qosConfig[pgProfile]["pkts_num_trig_pfc"]
         elif "wm_pg_shared_lossy" in pgProfile:
+            if not get_src_dst_asic_and_duts['single_asic_test'] and \
+                dutTestParams["basicParams"].get("platform_asic", None) \
```
Comment (abdosi) [inline tests/qos/test_qos_sai.py:1373]: Remove this

**Review by abdosi — CHANGES_REQUESTED** [review]
As comment

File: tests/qos/test_qos_sai.py (hunk 1 above)
Comment (rraghav-cisco) [reply tests/qos/test_qos_sai.py]: Changed it to: "This test needs to be revisited later, for the case where src and dst ASICs are different."

File: tests/qos/test_qos_sai.py [hunk 4]
```diff
@@ -1031,8 +1034,9 @@ def testQosSaiLossyQueueVoq(
             Raises:
                 RunAnsibleModuleFail if ptf test fails
         """
-        if dutTestParams["basicParams"]["sonic_asic_type"] != "cisco-8000":
-            pytest.skip("Lossy Queue Voq test is not supported")
+        if not get_src_dst_asic_and_duts['single_asic_test']:
+             pytest.skip("LossyQueueVoq: This test is skipped for now, will"
+                 " be re-enabled for RH cards.")
```
Comment (abdosi) [inline tests/qos/test_qos_sai.py:1039]: VOQ1: this one will change to validate fair voq so we should update skip reason accordingly.

**Review by abdosi — CHANGES_REQUESTED** [review]
as comment

File: tests/qos/test_qos_sai.py [hunk 5]
```diff
@@ -1559,6 +1570,12 @@ def testQosSaiQSharedWatermark(
                 qosConfig = dutQosConfig["param"][portSpeedCableLength]
             triggerDrop = qosConfig[queueProfile]["pkts_num_trig_ingr_drp"]
         else:
+            if not get_src_dst_asic_and_duts['single_asic_test'] and \
```
Comment (ansrajpu-git) [inline tests/qos/test_qos_sai.py:1573]: the fixture get_src_dst_asic_and_duts does not hold the value 'single_asic_test'. If the check is required to verify single_asic  test then below can be added 
 src_dut_index == dst_dut_index && src_asic_index == dst_asic_index

Comment (rraghav-cisco) [reply tests/qos/test_qos_sai.py:1573]: @ansrajpu-git : I added this key in the same PR.

**Comment by rraghav-cisco** [comment]
> As comment

Done.

**Review by abdosi — APPROVED** [review]

**Comment by abdosi** [comment]
@wangxin can you please help merge this

**Comment by XuChen-MSFT** [comment]
@rraghav-cisco @abdosi @ansrajpu-git , Can you help to cherry pick to master branch and 202305 branch?

+ @wangxin @StormLiangMS  for vis

**Pull request merged**